### PR TITLE
[ntuple] Add tutorial for framework usage

### LINKED ISSUE #16290: [ntuple] Provide tutorial for (envisioned) framework usage
It would be good to provide a tutorial how we (RNTuple developers) envision the APIs to be used by experiment frameworks. Relevant classes and methods that come to mind:
 * Creation of (bare) `RNTupleModel` and `RFieldToken`s (for parallel writing, see also https://github.com/root-project/root/issues/16236)
 * Creation of (multiple) `RNTupleWriter` / `RNTupleParallelWriter` when `Append`ing to a single `TFile`
 * For parallel writing, creation of `RNTupleFillContext` per thread or another granularity
 * Creation of (potentially multiple) `REntry`s and using `BindRawPtr()` to fill objects
 * Using `FillNoFlush()`, `RNTupleFillStatus::ShouldFlushCluster()`, `FlushColumns()` (see https://github.com/root-project/root/issues/16241) and `CommitCluster()` to reduce the time in critical sections

### PR DESCRIPTION
**Pull request by hahnjo**
Closes #16290

### CODE REVIEW

File: tutorials/v7/ntuple/ntpl014_framework.C [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+            // FlushCluster() will flush data to the underlying TFile, so it requires synchronization.
+            std::lock_guard g(fFileService.GetMutex());
+            fillContext.FlushCluster();
+         }
+      }
+   }
+};
+
+// A SerializingOutputter uses a sequential RNTupleWriter to append an RNTuple to a TFile and a std::mutex to
+// synchronize multiple threads.
```
Comment (jblomer) [inline tutorials/v7/ntuple/ntpl014_framework.C:154]: Maybe we could add a note about IMT (or enable it), to show that page compression can still be in parallel even with the sequential mode.

**Review by jblomer — APPROVED** [review]
Very nice, LGTM! Perhaps invite a framework developer to take a look.

File: tutorials/v7/ntuple/ntpl014_framework.C (hunk 1 above)
Comment (hahnjo) [reply tutorials/v7/ntuple/ntpl014_framework.C]: I thought about this and unfortunately I think this is not possible / not efficient with the implementation in this tutorial: A thread blocking to acquire a `std::mutex` cannot "help" the other thread that is currently in the critical section by executing its tasks. I added a note, also on the direct usage of `std::thread`.

**Comment by hahnjo** [comment]
Hi @Dr15Jones @makortel @siliataider, this PR aims to provide a tutorial how we (ROOT) envision frameworks could use the RNTuple API for writing. If you have some time, it would be great if you could check whether it makes sense and demonstrates the relevant use cases. The "framework code" is obviously very much simplified and the goal is not to code a full production framework, but I hope it's close enough to what the "real" experiment frameworks are doing. Thanks in advance!

**Pull request merged**